pkg/auth/providerrefresh/refresher.go

```diff
@@ -159,33 +167,56 @@ func (r *refresher) refreshAttributes(attribs *v3.UserAttribute) (*v3.UserAttrib
 		return nil, err
 	}
 
-	loginTokens = make(map[string][]*v3.Token)
-	derivedTokens = make(map[string][]*v3.Token)
+	loginTokens = make(map[string][]accessor.TokenAccessor)
+	derivedTokens = make(map[string][]accessor.TokenAccessor)
+
+	// List all tokens, v3 and ext.
+	// For ext tokens we actually filter for the user here.
+
+	allV3Tokens, err := r.tokenLister.List("", labels.Everything())
+	if err != nil {
+		return nil, err
+	}
 
-	allTokens, err := r.tokenLister.List("", labels.Everything())
+	allExtTokens, err := r.extTokenStore.ListForUser(user.Name)
 	if err != nil {
 		return nil, err
 	}
 
+	// Merge the separate lists into a unified set
+	allTokens := make([]accessor.TokenAccessor, 0, len(allV3Tokens)+len(allExtTokens.Items))
+	for _, token := range allV3Tokens {
+		allTokens = append(allTokens, token)
+	}
+	for _, eToken := range allExtTokens.Items {
+		ex := eToken
+		allTokens = append(allTokens, &ex)
```

Comment:
This code can be simplified, this was fixed in Go 1.22 https://go.dev/blog/loopvar-preview#the-fix

Reply:
commit 7f21c334bcfc2119a87a75bbec265eacd80e165d